src/fastcs/mapping.py

```diff
@@ -0,0 +1,51 @@
+from dataclasses import dataclass
+from typing import Callable, NamedTuple
+
+from .attributes import Attribute
+from .controller import BaseController, Controller
+from .cs_methods import MethodInfo
+
+MethodData = NamedTuple(
+    "MethodData", (("name", str), ("info", MethodInfo), ("method", Callable))
+)
```

Comment:
I'll have a look.

I suppose I feel uncomfortable with putting too much stuff into the MethodInfo object because it seems to duplicate information held elsewhere. I think in reality it's just moving some of the Mapping complexity out of mapping.py.

Reply:
This has been done, but because the Method object is attached to the _class_ and not the _instance_, you need to perform binding elsewhere. I think this is why I made that decision originally.

For now, I have done the binding in the backend, but it could easily trip us up. I have used `MethodType` for this binding; just search for that in the code and you'll see how it is done.

Reply:
This has been done, but because the Method object is attached to the _class_ and not the _instance_, you need to perform binding elsewhere. I think this is why I made that decision originally.

For now, I have done the binding in the backend, but it could easily trip us up. I have used `MethodType` for this binding; just search for that in the code and you'll see how it is done.

Reply:
I see what you mean. I think it is OK, although the builtin `MethodType` is pretty obscure to me and it has no docstring. It might be good to have a trivial wrapper with a better name and a docstring that creates the bound method. Just to get some explanation when hovering in an IDE.